### PR TITLE
FITS file table graph visualization plugin

### PR DESCRIPTION
**Pull request by francoismg**
This plugin goal is to enable astrophysicists to explore and plot FITS file table and binary table data directly inside Galaxy.

To achieve this goal the plugin main package astrovisjs https://github.com/esg-epfl-apc/astrojsvis relies on : 
 - fits-reader javascript library https://github.com/esg-epfl-apc/fits-reader
 - BokehJS graph library https://docs.bokeh.org/en/latest/docs/user_guide/advanced/bokehjs.html
 - D3 graph library https://d3js.org/

All packages are loaded with npm and the plugin is built using webpack

Plugin main features are : 
- Explore any FITS file headers
- Explore table and bintable data in table format
- Load local files when dataset relies on external files data
- Multiple files handling
- Automatic column creation based on header data
- Dynamically plot data using D3 or BokehJS from user choices (columns, ranges, error bars, log, .....)
- Error bars support
- Arithmetic columns creation

fits graph plugin 1

fits graph plugin 2

fits graph plugin columns

fits graph plugin main

## How to test the changes?
- [x] Instructions for manual testing are as follows:
  1. Load a FITS file containing at least one table or binary table HDU, example file available here https://www.astro.unige.ch/mmoda/gallery/data-products/1e-11451-6141-isgri-light-curve-61 at the very bottom of the page
  2. Launch plugin with selected file
  3. Select file from the "available files" column and use the "Add to plot" button at the bottom to switch it to "current files" column
  4. Select value for X Y axis and error bars if needed (for instance TIME, RATE, ERROR, TIMEDEL)
  5. Plot data using the "Generate" button

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by volodymyrss** [comment]
This plugin is made as part of EuroScienceGateway project in collaboration with @bgruening .

**Comment by guerler** [comment]
This looks good to me. However, we do not need a custom `mako` file anymore for these cases. Please take a look at and `cherry-pick` this commit if it works for you: 

https://github.com/guerler/galaxy/commit/cb40b6dc7ddbf090cf65701b7ee1863146b21223

It removes the mako and changes the visualization entry point to script. Everything else looks good.

**Comment by volodymyrss** [comment]
> This looks good to me. However, we don't need a custom `mako` file anymore for these cases. Please take a look at and `cherry-pick` this commit if it works for you:
> 
> [guerler@0eeee1c](https://github.com/guerler/galaxy/commit/0eeee1c6b5e61b268d7eddb799a7a32dd91db9bd)
> 
> It removes the mako and changes the visualization entry point to script. Everything else looks good.

@francoismg , does this work for you?

**Comment by francoismg** [comment]
> > This looks good to me. However, we don't need a custom `mako` file anymore for these cases. Please take a look at and `cherry-pick` this commit if it works for you:
> > [guerler@0eeee1c](https://github.com/guerler/galaxy/commit/0eeee1c6b5e61b268d7eddb799a7a32dd91db9bd)
> > It removes the mako and changes the visualization entry point to script. Everything else looks good.
> 
> @francoismg , does this work for you?

yes it seems good, I will try it locally and then make the changes

**Comment by francoismg** [comment]
Changes have been made and pushed

File: config/plugins/visualizations/fits_graph_viewer/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "fits-graph-visualization",
+    "version": "1.0.0",
+    "keywords": [
+        "galaxy",
+        "visualization"
+    ],
+    "main": "index.js",
+    "license": "AFL-3.0",
```
Comment (mr-c) [inline config/plugins/visualizations/fits_graph_viewer/package.json:9]: Shouldn't this be `MIT` and not `AFL-3.0`?

Comment (volodymyrss) [reply config/plugins/visualizations/fits_graph_viewer/package.json:9]: @francoismg , why did you select `AFL-3.0`?

Comment (mr-c) [reply config/plugins/visualizations/fits_graph_viewer/package.json:9]: As per the last sentence by @francoismg  in https://github.com/galaxyproject/galaxy/pull/19003#issue-2591306274

> * [x]  I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

 I'm changing this to `MIT` in https://github.com/galaxyproject/galaxy/pull/20264

**Pull request merged**